src/window_manager.rs

```diff
@@ -114,14 +118,55 @@ pub struct WindowManager {
     events: EventStream,
 }
 
+fn should_rename_after_event(event: Event) -> Result<()> {
+    match event {
+        Event::Workspace(boxed_workspace_event) => {
+            match (*boxed_workspace_event).change {
+                ch @ WorkspaceChange::Focus => {
+                    // Triggering on Init triggers an extra focus after
+                    // the initial focus.
+                    // Simply running a rename on every Workspace Focus
+                    // prevents this irritating visual artifact,
+                    // at the cost of running renames more often.
+                    info!("Renaming on WorkspaceEvent: {:#?}", ch);
+                    Ok(())
+                }
+                ch => {
+                    debug!("Rejected Rename for WorkspaceEvent: {:#?}", ch);
+                    Err(anyhow!("bad_ev")
+                        .context("Only certain WorkspaceChange events should trigger rename"))
+                }
+            }
+        }
+        Event::Window(boxed_window_event) => match (*boxed_window_event).change {
+            ch @ (WindowChange::New
+            | WindowChange::Close
+            | WindowChange::Title
+            | WindowChange::Move
+            | WindowChange::Urgent
+            | WindowChange::Mark) => {
+                info!("Renaming on WindowEvent: {:#?}", ch);
+                Ok(())
+            }
+            ch => {
+                debug!("Rejected Rename for WindowEvent: {:#?}", ch);
+                Err(anyhow!("bad_ev")
+                    .context("Only certain WindowChange events should trigger rename"))
+            }
+        },
+        _ => Err(anyhow!("bad_ev")
+            .context("Only certain Workspace and Window *Change events trigger a rename")),
+    }
+}
+
 impl WindowManager {
     pub fn connect() -> Result<Self> {
         Ok(Self {
             connection: Connection::new().context("Couldn't connect to WM")?,
             events: Connection::new()
                 .context("Couldn't connect to WM")?
-                .subscribe(&[EventType::Window])
-                .context("Couldn't subscribe to events of type Window")?,
+                .subscribe(&[EventType::Window, EventType::Workspace])
```

Comment:
So, this line alone is actually the **only** change that's _strictly_ required to allow for renaming empty workspaces. Every other change in this file isn't strictly needed.

There's a catch: Changing only this makes it so that _every single sway event_ that has anything at all to do with workspaces or windows will trigger a renaming operation of all workspaces.

I felt that was rather wasteful.